tests/framework/ab_test.py

```diff
@@ -270,35 +233,26 @@ def check_regression(
     )
 
 
-@contextlib.contextmanager
-def temporary_checkout(revision: str):
-    """
-    Context manager that checks out firecracker in a temporary directory and `chdir`s into it
+@with_filelock
+def git_clone(clone_path, commitish):
+    """Clone the repository at `commit`.
 
-    Will change back to whatever was the current directory when the context manager was entered, even if exceptions
-    happen along the way.
+    :return: the working copy directory.
     """
-    with TemporaryDirectory() as tmp_dir:
-        basename = Path(tmp_dir).name
+    if not clone_path.exists():
         ret, _, _ = utils.run_cmd(
-            f"git cat-file -t {revision}", ignore_return_code=True
+            f"git cat-file -t {commitish}", ignore_return_code=True
         )
         if ret != 0:
-            # git didn't recognize this object, so maybe it is a branch; qualify it
-            revision = f"origin/{revision}"
+            # git didn't recognize this object; qualify it if it is a branch
+            commitish = f"origin/{commitish}"
         # make a temp branch for that commit so we can directly check it out
-        utils.run_cmd(f"git branch {basename} {revision}")
-        # `git clone` can take a path instead of an URL, which causes it to create a copy of the
-        # repository at the given path. However, that path needs to point to the root of a repository,
-        # it cannot be some arbitrary subdirectory. Therefore:
+        utils.run_cmd(f"git branch {clone_path} {commitish}")
         _, git_root, _ = utils.run_cmd("git rev-parse --show-toplevel")
         # split off the '\n' at the end of the stdout
-        utils.run_cmd(f"git clone -b {basename} {git_root.strip()} {tmp_dir}")
-        yield Path(tmp_dir)
-
-        # If we compiled firecracker inside the checkout, python's recursive shutil.rmdir will
-        # run incredibly long. Thus, remove manually.
-        utils.run_cmd(f"rm -rf {tmp_dir}")
+        utils.run_cmd(f"git clone -b {clone_path} {git_root.strip()} {clone_path}")
+    utils.run_cmd("./tools/release.sh --profile release", cwd=clone_path)
```

Comment:
Mh, the only problem I see with this is that it will also compile Firecracker for A/B-Tests that do not need a compiled Firecracker (e.g. the cargo audit A/B-tests, or anything that uses `git_ab_test_host_command` and friends). I don't think its a big problem practically (as all A/B-Tests of a PR will use the same two commits, and the compiled binaries are reused, so whether the binary gets compiled by a test that doesnt use it doesnt matter because a later test will use it), but it might be worth to keep something like `git_ab_test_with_binaries` around and have it decorate the `test_runner` function with something that runs `release.sh`? Mainly for when subsets of A/B-tests are ran locally.

Reply:
Done